app/assets/v2/js/quadraticlands/mission/diplomacy/room.js

```diff
@@ -0,0 +1,175 @@
+        leave_room_interface.classList.add('warning');
+      } else {
+        leave_room_button.classList.add('disabled');
+        leave_room_interface.classList.remove('warning');
+      }
+    });
+
+  }
+
+
+  // vouche button trigger function vouche()
+  //
+  const vouche_button = document.getElementById('vouche_button');
+  vouche_button.addEventListener('click', () => {
+    vouche();
+  });
+
+
+  // vouche bar validations
+  inputs = document.querySelectorAll('.member input');
+  inputs.forEach(i => {
+    i.addEventListener('input', input => {
+      if (i.value < 0) {
+        i.value = 0;
+      }
+
+      updateVoucheBar();
+
+      console.debug("USED", window.used)
+      console.debug("USE", window.use)
+      console.debug("AWAILABLE", (window.walletbalance - window.used) )
+
+      if (window.use > (window.walletbalance - window.used) )
+      {
+        console.debug("you can not use so much gtc");
+        i.value = 0;
+        updateVoucheBar();
+      }
+
+    });
+  });
+
+});
+
+
+function updateVoucheBar(){
+
+  window.use = 0;
+  inputs = document.querySelectorAll('.member input');
+
+  inputs.forEach(i => {
+    if (i.value > 0) {
+      window.use += Number(i.value);
+    }
+  });
+
+  diplomacy_wallet_use = document.getElementById('diplomacy_wallet_use');
+  diplomacy_wallet_use.innerHTML = window.use;
+
+  diplomacy_wallet_available = document.getElementById('diplomacy_wallet_available');
+  diplomacy_wallet_available.innerHTML = window.walletbalance - window.used;
+  console.debug("updateVouceBar");
+
+}
+
+// fetch the gtc balance
+async function diplomacyWallet(){
+  console.debug("diplomacy wallet");
+  try {
+
+    let balance = await getTokenBalances(gtc_address());
+
+    diplomacy_wallet_balance.innerHTML = balance.balance.toFixed(2);
+    diplomacy_wallet_address.innerHTML = truncate(selectedAccount);
+    window.walletbalance = balance.balance.toFixed(2);
+    console.debug("walletbalance", window.walletbalance)
+
+    // to calc available balance
+    updateVoucheBar();
+
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+
+// reads all the members input fields and generate a nice
+// array of objects to do a "sign" + a "safe to db"
+// not sure how to do this part.
+
+async function vouche() {
+
+  console.debug('VOUCHE');
+
+  // read all input fields what have data-member
+  members = document.querySelectorAll('[data-member]');
+
+  // define a result array
+  const result = [];
+
+  // push all vouche data of each member to the result
+  members.forEach(member => {
+    var entry = {
+      userid: member.dataset.userid,
+      username: member.dataset.username,
+      value: member.value
+    };
+
+    result.push(entry);
+  });
+
+  // @kev : do something with the result ( sign, safe, whatever)
+  // this is how far i could come. now your turn :)
+  const accounts = await web3.eth.getAccounts();
+  let signature = web3.eth.sign(JSON.stringify(result), accounts[0], function(foo){
```

Comment:
i cant get the signature back for some reason. i dont understand why

Reply:
@owocki `web3.eth.sign` seems to be outdated which is why the promise gets stuck forever
Here give this a shot 

`let signature = await web3.eth.personal.sign(JSON.stringify(result) , accounts[0]);`

This works when I try it out

![image](https://user-images.githubusercontent.com/5358146/121666523-f9e80580-cac6-11eb-9470-597a5346585e.png)

Reply:
this worked thx